activerecord/lib/active_record/connection_adapters/abstract/quoting.rb

```diff
@@ -17,7 +22,11 @@ def quote(value)
       # SQLite does not understand dates, so this method will convert a Date
       # to a String.
       def type_cast(value, column = nil)
+        value = id_value_for_database(value) if value.is_a?(Base)
```

Comment:
If `value` is an AR::Base object, using `value.id` has potentially a bug.
If `id: :datetime, precision: 3`, typed cast value should be `2017-02-14 12:34:56.789000` by `quoted_date`, not `2017-02-14 12:34:56 +0000`.